CMakeLists.txt

```diff
@@ -232,9 +232,11 @@ endif()
 include(DownloadBcrypt)
 
 target_include_directories(eoserv PUBLIC ${CMAKE_BINARY_DIR}/bcrypt-src/include)
+target_include_directories(eoserv_lib PUBLIC ${CMAKE_BINARY_DIR}/bcrypt-src/include)
 list(APPEND eoserv_LIBRARIES bcrypt)
 
-target_link_libraries(eoserv ${eoserv_LIBRARIES})
+target_link_libraries(eoserv eoserv_lib ${eoserv_LIBRARIES})
+target_link_libraries(eoserv_lib ${eoserv_LIBRARIES})
```

Comment:
You can eliminate the `${eoserv_LIBRARIES}` from line 238 since they are transitively linked by the link established on line 239.

```cmake
target_link_libraries(eoserv eoserv_lib)
target_link_libraries(eoserv_lib ${eoserv_LIBRARIES})
```